### PR TITLE
Reactive Component Bugfixes on the Falstad Importer

### PR DESCRIPTION
**Pull request by Grissess**
This PR contains some important changes to the MNA that should fix at least the reactive components--these regressions were discovered when the Falstad importer was used on the "basic LRC" circuit that [Falstad's simulator](http://www.falstad.com/circuit/) initially loads.

### CODE REVIEW

File: core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt [hunk 1]
```diff
@@ -130,15 +133,14 @@ class Falstad(val source: String) {
 		if(DEBUG) {
 			val repmap: MutableMap<PosSet, MutableSet<PosSet>> = mutableMapOf()
 			roots.values.forEach {
-				dprintln("F.<init>: r $it => ${it.representative}")
+				dprintln("F.<init>: r ${it} => ${it.representative}")
```
Comment (jrddunbr) [inline core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt:136]: need to revert this change

File: core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt [hunk 2]
```diff
@@ -130,15 +133,14 @@ class Falstad(val source: String) {
...
 				repmap.getOrPut(it.representative as PosSet, { mutableSetOf() }).add(it)
 			}
 
 			repmap.entries.forEach {
 				dprintln("F.<init>: R ${it.key}:")
 				it.value.forEach {
-					it2 ->
-					dprintln(" - $it2")
+					dprintln(" - $it")
```
Comment (jrddunbr) [inline core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt:143]: revert

File: core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt [hunk 3]
```diff
@@ -158,11 +160,10 @@ class Falstad(val source: String) {
...
 			(0 until ordered.size - 1).forEach {
-					it2 ->
-					ordered[it2].component.connect(
-					ordered[it2].pinidx,
-					ordered[it2+1].component,
-					ordered[it2+1].pinidx
+				ordered[it].component.connect(
+					ordered[it].pinidx,
+					ordered[it+1].component,
+					ordered[it+1].pinidx
```
Comment (jrddunbr) [inline core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt:166]: revert

File: core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt (hunk 2 above)
Comment (jrddunbr) [reply core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt:143]: This change is because the `it` variable shadowed the larger scope's variable, and I've seen that explode in fantastically nondeterministic ways at least in my experience.

File: core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt (hunk 1 above)
Comment (Grissess) [reply core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt:136]: The change here involves adding braces to a format string; this actually triggers a lint in IDEA ("Remove unnecessary braces"). Are you sure about this one?

Also, it's in the debug path, not at all hotpath or maintainability target.

File: core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt (hunk 2 above)
Comment (Grissess) [reply core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt:143]: It very much shouldn't explode; the lexical scoping here should be sensible. One does have to take care to note that the outer `it` is indeed inaccessible, or change the name via a named parameter.

If the change were a good idea, I'd certainly prefer a more descriptive name than `it2`.

File: core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt (hunk 3 above)
Comment (Grissess) [reply core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt:166]: idem, as above.

File: core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt (hunk 1 above)
Comment (jrddunbr) [reply core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt:136]: The git changes seem to be reverting the change, the braces are not needed (as IDEA says)

File: core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt (hunk 2 above)
Comment (jrddunbr) [reply core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt:143]: How about we just avoid the problem entirely and name it properly? I couldn't come up with a good name at the time.

File: core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt (hunk 1 above)
Comment (Grissess) [reply core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt:136]: I'm dumb, I read the diff in the wrong order :)
